### ISSUE TITLE
Subsite switching: open page appears to move along to new subsite

### ISSUE THREAD

**Issue by micschk** [silverstripe/silverstripe-subsites#489]
When switching to another subsite while having a specific page open in edit mode, that page will be re-opened in the other subsite as if it has move there. This is confusing and it seems there are also [some](https://github.com/silverstripe/silverstripe-subsites/issues/404) [other](https://github.com/silverstripe/silverstripe-subsites/issues/461) issues related to this behaviour. It seems the system just finds the page by its ID and opens it in edit mode (even though it should be filtered out because it belongs to another subsite).

I propose we remove the specific ID (of any currently edited file/folder/page/settings/history/etc) when switching subsites. That way you'd always start with an overview of all pages/files/etc in the subsite you've just switched to.

```JS
                    // Switch to listing views of subsite-filtered sections (eg: pages/ & assets/)
                    let newLocHref = window.location.href
                        .split('edit/show/').shift() // remove [pages/]edit/etc
                        .split('settings/show/').shift() // remove [pages/]settings/etc
                        .split('history/show/').shift() // remove [pages/]history/etc
                        .split('show/').shift(); // remove [assets/]show/102/edit/etc (last as 'show' is broad)
                    window.location.href = newLocHref + $.query.set('SubsiteID', $(this).val()).toString();
```
PR coming up.

**Comment by lekoala** [comment]
somehow related, I have a similar issue in modeladmin when switching subsite while editing a record, the record can stay displayed even though you changed the subsite (which can cause lot of isuses).

fixed this by doing this beforeInit

            if ($currentController instanceof ModelAdmin) {
                // We are editing something, move to root
                if (strpos($request->getURL(), '/EditForm/') !== false) {
                    return $this->owner->redirect($currentController->Link());
                }
            }

Happy to make a PR if that seems like a solution you would consider, but it's probably not great to do all kind of controller specific logic.
Basically, I think when switching subsite, we should always move to the root of the current section if available, or to the default segment.

**Comment by GuySartorelli** [comment]
Fixed in https://github.com/silverstripe/silverstripe-subsites/pull/490. It will be tagged automatically when CI has finished running on the branch.

**Comment by GuySartorelli** [comment]
The fix needed to be reverted - see https://github.com/silverstripe/silverstripe-subsites/issues/538

**Comment by GuySartorelli** [comment]
@emteknetnz you've just closed this - I think based on the way GitHub references that, it was an accidental automatic closing. Can you please confirm?

**Comment by emteknetnz** [comment]
Yeah accidental. I was fixing broken merge-ups at the time

**Comment by adiwidjaja** [comment]
This is no longer the case in the current version 4.0.1, the issue can be closed.
